### ISSUE TITLE
Changelog info text in F-droid

### ISSUE THREAD

**Issue by aha999** [streetcomplete/StreetComplete#3355]
**Use case**

Currently, there is no visible window with the latest changes when you open your app's page in F-Droid.

**Proposed Solution**

Add changelog text when you open the fdroid app repo

**Some tips:**

> F-Droid looks for metadata in the app’s source repo. The supported formats for this metadata are borrowed from popular projects that automate uploading this metadata to Google Play Store.

>The one in your example corresponds to Fastlane Supply and is documented [here](https://docs.fastlane.tools/actions/supply/#changelogs-whats-new). All formats supported by F-Droid are listed [here](https://f-droid.org/en/docs/All_About_Descriptions_Graphics_and_Screenshots/#in-the-applications-source-repository)

**Comment by matkoniecz** [comment]
I think that it is a bug caused by F-Droid and should be reported on F-Droid repository.

Is there some reason to suspect that it is bug caused by StreetComplete itself? This works well when installed from Google Play, it works well when sideloaded.

Is F-Droid modifying StreetComplete in some way before releasing it?

**Comment by aha999** [comment]
Maybe @eighthave would know how to answer this ?

**Comment by FloEdelmann** [comment]
Wo could do it the other way around: Manually create one file per version in that location, and then combine them into the all-in-one file with a Gradle task.

**Comment by westnordost** [comment]
And how to get from version code to version name then?

**Comment by FloEdelmann** [comment]
Just leave it as a HTML heading in the individual files.

**Comment by westnordost** [comment]
Hmm, that would be possible. I wonder what F-Droid does with `*.txt` files that are actually html (snippets) though.

**Comment by westnordost** [comment]
Also, do you know an example app for which this changelog stuff is used? I would like to see how it is displayed because I am still not sure if this is worth it at all.

**Comment by FloEdelmann** [comment]
HTML Markup seems to be possible in at least the app description: https://forum.f-droid.org/t/f-droid-support-fastlane-full-description-in-html-or-md-instead-of-txt/9395

**Comment by westnordost** [comment]
What exactly are you planning to do and why?

**Comment by westnordost** [comment]
> Imo markdown is the best option here. It's the easiest to update and copy-paste to different places (e.g. GitHub releases)

<h3>This</h3> is <i>HTML</i> and it
<ul>
<li>works</li>
</ul>
no problem

**Comment by westnordost** [comment]
If the changelog was in markdown, we'd need to include a library that can display a markdown-formatted string. Currently,[ we use a library to display simple html](https://github.com/SufficientlySecure/html-textview).

**Comment by aha999** [comment]
> Mh well, that would be possible. @aha999, do you have any information on how F-Droid treats HTML? Or if it supports HTML, markup or something else at all beside `.txt`?

i do not. I only know that a lot of projects use this feature to have changelogs displayed but if other formats are allowable i would not know that. Maybe @eighthave has an answer to this ?

**Comment by tiptoptom** [comment]
Do I understand correctly that now that we have the changelog as .md, we only need the version number and could then build an automatism?

**Comment by FloEdelmann** [comment]
Yeah, basically. But linking each version code to a Markdown heading might not be that easy.

**Comment by tiptoptom** [comment]
Unfortunately, I really don't know how this works with the tasks. Where is the version code defined? And if you have the version code and version name, can't you regex it?

Alternatively, if that makes it easier, I think you could always copy the entire changelog and add the current version code.  
https://github.com/M66B/FairEmail/tree/master/metadata/en-US/changelogs

**Comment by FloEdelmann** [comment]
> I really don't know how this works with the tasks

They are defined in https://github.com/streetcomplete/StreetComplete/tree/master/buildSrc and registered in https://github.com/streetcomplete/StreetComplete/blob/master/build.gradle.kts and https://github.com/streetcomplete/StreetComplete/blob/master/app/build.gradle.kts. You can execute them with e.g. `./gradlew generateQuestList` or `./gradlew test`.

> Where is the version code defined?

In [`f7c0cc61`/app/build.gradle.kts#L37](https://github.com/streetcomplete/StreetComplete/blob/f7c0cc61a3d6f37fe2bea6fa5242d741d2aa463e/app/build.gradle.kts#L37). Since there is also the version name there, matching could indeed be feasible.

> And if you have the version code and version name, can't you regex it?

Yeah, that's basically what happens in the changelog dialog in the app as well, see [`f7c0cc61`/app/src/main/java/de/westnordost/streetcomplete/screens/about/ChangelogFragment.kt#L56-L64](https://github.com/streetcomplete/StreetComplete/blob/f7c0cc61a3d6f37fe2bea6fa5242d741d2aa463e/app/src/main/java/de/westnordost/streetcomplete/screens/about/ChangelogFragment.kt#L56-L64).

**Comment by scollovati** [comment]
For some reasons here it appears: https://f-droid.org/en/packages/de.westnordost.streetcomplete.expert/ instead in the basic version does not work: https://f-droid.org/en/packages/de.westnordost.streetcomplete/

**Comment by scollovati** [comment]
> You can see SCEE versions that have a changelog here: https://github.com/Helium314/SCEE/tree/modified/metadata/en-US/changelogs

Exactly. I was wondering why the base project do not make it available for F-Droid users.
